stock_location_bin_name/models/stock_location.py

```diff
@@ -37,14 +38,25 @@ class StockLocation(models.Model):
     _inherit = "stock.location"
 
     location_name_format = fields.Char(
-        "Location Name Format",
         help="Format string that will compute the name of the location. "
         "Use location fields. Example: "
         "'{area}-{corridor:0>2}.{rack:0>3}"
         ".{level:0>2}'\n"
         "Missing fields are replaced by '~' and formatting errors by '!'.",
     )
 
+    @api.constrains("location_name_format")
+    def _check_location_name_format(self):
+        for loc in self:
+            if loc.location_kind == "bin" and loc.location_name_format:
+                raise ValidationError(
+                    self.env._(
+                        "The area of the location: '%(location)s' "
+                        "should have the name format, not in a Bin location",
+                        location=loc.name,
+                    )
```

Comment:
This looks a little bit too specific and at the same time a little vague.
Location Name Format could be set on the Area or on the Zone, as long as it's not a bin.
I would be more direct about what the issue is.
```suggestion
                    self.env._(
                        "Location Name Format cannot be set on 'bin' type locations."
                    )
```